### PR TITLE
chore(ci): fix failure condition when modified files are empty

### PR DESCRIPTION
**Pull request by HansRobo**
## Description

 Fix failure condition when modified files are empty in `cppcheck-differential`.

## Related links

A test due to empty modified files. (the PR deletes all the package files)
https://github.com/autowarefoundation/autoware.universe/actions/runs/9673888348/job/26688603324 

## How was this PR tested?

The workflow works well with diffs that includes deleted files only. 
https://github.com/HansRobo/autoware.universe/actions/runs/9675016733/job/26691735752

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: .github/workflows/cppcheck-differential.yaml [hunk 1]
```diff
@@ -31,21 +31,20 @@ jobs:
...
-        id: changed-files
+      - name: Get modified files
+        id: get-modified-files
         run: |
-          git diff --name-only "origin/${{ github.base_ref }}"...HEAD > changed_files.txt
-          cat changed_files.txt
+          echo "changed_files=$(git diff --name-only "origin/${{ github.base_ref }}"...HEAD | grep -E '\.(cpp|hpp)$' | tr '\n' ' ' || true)" >> $GITHUB_OUTPUT
         shell: bash
-
+
```
Comment (HansRobo) [inline .github/workflows/cppcheck-differential.yaml:39]:
```suggestion
```

**Comment by veqcc** [comment]
@HansRobo 
Is there any difference between `modified` and `changed` ?

**Comment by HansRobo** [comment]
@veqcc 

> Is there any difference between modified and changed ?

No.
The spelling variations between `changed` and `modified` have been unified to `changed` in https://github.com/autowarefoundation/autoware.universe/pull/7704/commits/6a8989890682a9e12e1fa5a48af2ea63a1ac899e.

**Review by veqcc — APPROVED** [review]
I really appreciate your quick work!! LGTM :+1:

**Comment by veqcc** [comment]
@HansRobo 
Please confirm DCO!!

**Comment by HansRobo** [comment]
@veqcc I set DCO to pass manually.

**Pull request merged**